runtime/runtime.m

```diff
@@ -270,8 +270,10 @@
 	if (index == -1) {
 		p = mono_signature_get_return_type (msig);
 	} else {
-		for (int i = 0; i < index + 1; i++)
+		for (int i = 0; i < index + 1; i++) {
+			xamarin_mono_object_release (&p);
 			p = mono_signature_get_params (msig, &iter);
+		}
```

Comment:
what will release the last `p` assigned in the loop ?

Reply:
It will be returned from this function, and the caller will have to release it.